### DISCUSSION TITLE
Сумма из префильтра

### DISCUSSION THREAD

**Discussion by Makhmut2445** [totumonline/totum-ru-issues-and-discussions#97] [Как это сделать?] [answered]
Как сделать, чтоб суммарное значение всего поля менялась в зависимости от выбранных строк в префильтре?
Поле суммы: Хедер, отображается в Футере.
Таблица Цикловая.
![Telegram_xsoMvdZY5A](https://user-images.githubusercontent.com/103171127/162136043-ea19f630-3017-4d1b-8e94-7c29a85e5b69.png)

**Comment by erpic-ru** [comment]
Добавляете нередактируемое числовое денежное поле Total in stock в префильтры последним по порядку.

Добавляете код. В where прописываете все фильтры, которые намерены использовать:
```

= : listSum(list: $list)

list: selectList(table: $#ntn; field: 'summ'; where: 'name' = #fl_name; where: 'brand' = #fl_brand; where....)
```
Возможно, есть более удобный способ

**Reply by Makhmut2445** [reply]
Префильтр установлен только на поле Brand. Возвращает значение 0 вместо суммы.

**Reply by erpic-ru** [reply]
Поле с суммой должно стоять по порядку (sort) после бренда в префильтре. В поисковом поле префильтра (в вашем случае brand) установлено значение по умолчанию `*ALL*`

Все работает даже без использования кнопки применения префильтра

**Reply by Makhmut2445** [reply]
![image](https://user-images.githubusercontent.com/103171127/162194999-0e822976-bd59-4ecb-bcd8-710db30a3d5d.png)

Например.

**Reply by Makhmut2445** [reply]
Вот тут что не правильно?
Стоит код такой:
![image](https://user-images.githubusercontent.com/103171127/162195400-a4e90807-5d6f-418f-9c0f-09570dcc5da9.png)

**Reply by Makhmut2445** [reply]
Получилось сделать. Спасибо.)

**Comment by ttm-alexeypolunin** [comment, marked as answer]
### [Видео на YouTube —>](https://youtu.be/i5ERdtNJ5nQ)

Таблица Циклов в этом месте работает аналогично Простой, показанной в видео.

Код в поле в форматировании:

```
f1=: setFormat(text: $format)

format: numFormat(num: $calc; dectimals: 2; decsep: "."; thousandssep: ","; prefix: "$ ")

calc: listSum(list: $list)

list: selectList(table: $#ntn; field: 'summ'; where: 'brand' = #fl_brand)
```

Если в префильтре мультиселект:

**UPD: с 3.7.49 неактуально так-как список из одного значения `["*ALL*"]` будет обработан так же как и `"*ALL*"`.**

```
f1=: setFormat(text: $format)

format: numFormat(num: $calc; dectimals: 2; decsep: "."; thousandssep: ","; prefix: "$ ")

calc: listSum(list: $list)

list: selectList(table: $#ntn; field: 'summ'; where: 'brand' = $if)

	if: if(condition: #fl_brand[0] = "*ALL*"; then: "*ALL*"; else: #fl_brand)
```
